pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -533,13 +533,12 @@ func (t *CycleNodeRequestTransitioner) transitionHealing() (reconcile.Result, er
 
 		// try and re-attach the nodes, if any were un-attached
 		t.rm.LogEvent(t.cycleNodeRequest, "AttachingNodes", "Attaching instances to nodes group: %v", node.Name)
-		alreadyAttached, err := nodeGroups.AttachInstance(node.ProviderID, node.NodeGroupName)
-		if alreadyAttached {
+		// if the node is already attached, ignore the error and continue to un-cordoning, otherwise return with error
+		if alreadyAttached, err := nodeGroups.AttachInstance(node.ProviderID, node.NodeGroupName); alreadyAttached {
```

Comment:
Personally, I find this a little more confusing than what we had before. I think we can still achieve what you're wanting to achieve by not putting the `alreadyAttached, err := nodeGroups.AttachInstance(node.ProviderID, node.NodeGroupName)` into an if statement, but by just adjusting the if statements

Reply:
What do you think it updating `nodeGroups.AttachInstance` instead by not returning the error. It was already deemed as being an error that we expected so it would also just not be returned.

Reply:
I did think about that, it uses `verifyIfErrorOccurred` to check and return error - which is also used by `DetachInstance`, so I didn't do it. 

Just updated as @awprice suggested, it looks better now I think.